Bio/Nexus/Trees.py

```diff
@@ -86,48 +86,65 @@ def _parse(self, tree):
         # Remove any leading/trailing white space - want any string starting
         # with " (..." should be recognised as a leaf, "(..."
         tree = tree.strip()
+
         if tree.count("(") != tree.count(")"):
             raise TreeError("Parentheses do not match in (sub)tree: " + tree)
         if tree.count("(") == 0:  # a leaf
             # check if there's a colon, or a special comment, or both  after the taxon name
             nodecomment = tree.find(NODECOMMENT_START)
             colon = tree.find(":")
             if colon == -1 and nodecomment == -1:  # none
-                return [tree, [None]]
+                branch, comment = tree, [None]
             elif colon == -1 and nodecomment > -1:  # only special comment
-                return [tree[:nodecomment], self._get_values(tree[nodecomment:])]
+                branch, comment = tree[:nodecomment], self._get_values(
+                    tree[nodecomment:]
+                )
             elif colon > -1 and nodecomment == -1:  # only numerical values
-                return [tree[:colon], self._get_values(tree[colon + 1 :])]
+                branch, comment = tree[:colon], self._get_values(tree[colon + 1 :])
             elif (
                 colon < nodecomment
             ):  # taxon name ends at first colon or with special comment
-                return [tree[:colon], self._get_values(tree[colon + 1 :])]
+                branch, comment = tree[:colon], self._get_values(tree[colon + 1 :])
             else:
-                return [tree[:nodecomment], self._get_values(tree[nodecomment:])]
+                branch, comment = tree[:nodecomment], self._get_values(
+                    tree[nodecomment:]
+                )
+
+            return [branch, comment]
         else:
             closing = tree.rfind(")")
             val = self._get_values(tree[closing + 1 :])
             if not val:
                 val = [None]
+
             subtrees = []
             plevel = 0
             prev = 1
             incomment = False
-            for p in range(1, closing):
-                if not incomment and tree[p] == "(":
-                    plevel += 1
-                elif not incomment and tree[p] == ")":
-                    plevel -= 1
-                elif tree[p:].startswith(NODECOMMENT_START):
-                    incomment = True
-                elif incomment and tree[p] == NODECOMMENT_END:
-                    incomment = False
-                elif not incomment and tree[p] == "," and plevel == 0:
-                    subtrees.append(tree[prev:p])
-                    prev = p + 1
 
+            sep = re.compile(
```

Comment:
Maybe compile the regex once at top level, perhaps as a private module level variable?

Reply:
What name would be a good choice? `_BLOCK_SEPARATOR`?

Reply:
I'd expect something in there about comments, maybe ``_re_block_comment_separator``?

(I'd use UPPER for a constant, which I guess a compiled regex kinda is?)

Looking at ``SeqFeature.py`` I've not consistently used ``_re_...`` as a prefix there.

Reply:
>I'd expect something in there about comments, maybe _re_block_comment_separator?

Since the regex acts in the general sense (commas and parenthesis are also considered as block delimiters) I'm not sure mentioning a comment would be helpful.

>Looking at SeqFeature.py I've not consistently used _re_... as a prefix there.

For consistency, what about `_re_block_delimiters`?